datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -639,15 +614,47 @@ fn write_sorted(
     Ok(())
 }
 
-fn read_spill(sender: Sender<Result<RecordBatch>>, path: &Path) -> Result<()> {
+/// Stream batches from spill files.
+///
+/// Each spill file has one or more batches. Intra-batch order is guaranteed (each one is sorted),
+/// but the inter-batch ordering is not guaranteed, hence why we need to convert each batch from the
+/// spill to a separate input stream for the merge-sort procedure.
```

Comment:
Will this produce unnecessary comparison when the inter-batch ordering is guaranteed for the normal case without `K`?

Reply:
I think not, because than it should be single record batch per spill. I can add a sort counter to double-check though.

Reply:
A spill file is generated by `in_mem_sort()` and then spill. And `spill_sorted_batches` takes `batches: Vec<RecordBatch>` as one of the arguments.

Reply:
Ah, good point; for some reason I thought that only 1 batch gets collected from `in_mem_sort_stream` in case of no fetch, even though that is not a given.